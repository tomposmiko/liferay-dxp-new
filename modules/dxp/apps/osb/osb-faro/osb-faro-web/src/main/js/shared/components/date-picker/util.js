import moment from 'moment';
import {conformsTo, isNil} from 'lodash';

/**
 * Represents a time between two dates.
 * @typedef Range
 * @property {Moment} start - The start of the range.
 * @property {Moment} end - The end of the range.
 */

export function isAboveMaxRange(date, maxRange) {
	return (
		isRange(date) &&
		date.start &&
		date.end &&
		maxRange &&
		moment
			.duration({
				from: date.start,
				to: date.end
			})
			.asDays() > maxRange
	);
}

/**
 * A validation function that determines if a value is a valid
 * moment instance, allowing for null.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
function isMomentOrNil(value) {
	return isNil(value) || moment.isMoment(value);
}

/**
 * A validation function that determines if the value is a valid
 * moment or range.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
export function isDateOrRange(value) {
	return isNil(value) || moment.isMoment(value) || isRange(value);
}

/**
 * Determines if a date is in a {@link Range}.
 * @param {Range|Moment} dateOrRange - The range to check in.
 * @param {Moment} date - The date to check for.
 * @returns {boolean}
 */
export function isInRange(dateOrRange, date) {
	if (moment.isMoment(dateOrRange)) {
		return false;
	}

	const {end, start} = dateOrRange;

	return start && date.isAfter(start) && end && date.isBefore(end);
}

/**
 * A validation function that determines if a value is a valid {@link Range}.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
export function isRange(value) {
	return conformsTo(value, {
		end: isMomentOrNil,
		start: isMomentOrNil
	});
}

/**
 * Adds a date to an existing {@link Range}.
 *
 * This function will try to be smart and first update the
 * start or end if either are null. If both exist, then it will return
 * a new range with the start set to the new date, and the end reset
 * to null.
 * @param {Range} range - The range to update.
 * @param {Moment} date - The date to add to the range.
 * @returns {Range}
 */
export function updateRange(range, date) {
	if (!range.start) {
		return {
			...range,
			start: date
		};
	}

	if (range.start && !range.end) {
		if (date.isSameOrAfter(range.start)) {
			return {
				...range,
				end: date
			};
		} else {
			return {
				end: range.start,
				start: date
			};
		}
	}

	if (date.isSame(range.start) && date.isSame(range.end)) {
		return range;
	}

	return {
		end: null,
		start: date
	};
}
