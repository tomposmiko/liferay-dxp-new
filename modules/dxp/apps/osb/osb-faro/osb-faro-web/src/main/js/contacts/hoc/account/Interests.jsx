import Card from 'shared/components/Card';
import getInterestsQuery from 'contacts/queries/InterestsQuery';
import React from 'react';
import URLConstants from 'shared/util/url-constants';
import {compositionListColumns} from 'shared/util/table-columns';
import {CompositionTypes} from 'shared/util/constants';
import {COUNT, createOrderIOMap} from 'shared/util/pagination';
import {
	getMapResultToProps,
	mapPropsToOptions
} from 'contacts/hoc/mappers/interests-query';
import {graphql} from '@apollo/react-hoc';
import {Routes, toRoute} from 'shared/util/router';
import {Sizes} from 'shared/util/constants';
import {useParams} from 'react-router-dom';
import {useQueryPagination} from 'shared/hooks';
import {withBaseResults} from 'shared/hoc';

const withData = () =>
	graphql(getInterestsQuery(CompositionTypes.AccountInterests), {
		options: mapPropsToOptions,
		props: getMapResultToProps(CompositionTypes.AccountInterests)
	});

const TableWithData = withBaseResults(withData, {
	defaultOrderByField: 'count',
	emptyDescription: (
		<>
			<span className='mr-1'>
				{Liferay.Language.get(
					'check-back-later-to-verify-if-data-has-been-received-from-your-data-sources'
				)}
			</span>

			<a
				href={URLConstants.AccountIndividualsDocumentationLink}
				key='DOCUMENTATION'
				target='_blank'
			>
				{Liferay.Language.get('learn-more-about-interests')}
			</a>
		</>
	),
	emptyIcon: {
		border: false,
		size: Sizes.XXXLarge,
		symbol: 'ac-satellite'
	},
	emptyTitle: Liferay.Language.get('there-are-no-interests-found'),
	getColumns: ({channelId, groupId, id, maxCount, totalCount}) => [
		compositionListColumns.getName({
			label: Liferay.Language.get('topic'),
			routeFn: ({data: {name}}) =>
				name &&
				toRoute(Routes.CONTACTS_ACCOUNT_INTEREST_DETAILS, {
					channelId,
					groupId,
					id,
					interestId: name
				}),
			sortable: true
		}),
		compositionListColumns.getRelativeMetricBar({
			label: Liferay.Language.get('account-members'),
			maxCount,
			sortable: true,
			totalCount
		}),
		compositionListColumns.getPercentOf({
			metricName: Liferay.Language.get('active-members'),
			totalCount
		})
	],
	rowIdentifier: 'name',
	showDropdownRangeKey: false
});

const Interests = () => {
	const {channelId, groupId, id} = useParams();
	const {delta, orderIOMap, page, query} = useQueryPagination({
		initialOrderIOMap: createOrderIOMap(COUNT)
	});

	return (
		<Card pageDisplay>
			<Card.Header className='align-items-center d-flex justify-content-between'>
				<Card.Title>
					{Liferay.Language.get('account-interest-topics')}
				</Card.Title>
			</Card.Header>

			<TableWithData
				channelId={channelId}
				delta={delta}
				groupId={groupId}
				id={id}
				orderIOMap={orderIOMap}
				page={page}
				query={query}
				rowBordered={false}
			/>
		</Card>
	);
};

export default Interests;
