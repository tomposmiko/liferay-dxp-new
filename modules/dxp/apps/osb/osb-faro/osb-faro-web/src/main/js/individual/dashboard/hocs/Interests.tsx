import Card from 'shared/components/Card';
import IndividualInterestsQuery from 'shared/queries/IndividualInterestsQuery';
import NoResultsDisplay from 'shared/components/NoResultsDisplay';
import React from 'react';
import URLConstants from 'shared/util/url-constants';
import {compositionListColumns} from 'shared/util/table-columns';
import {CompositionTypes, Sizes} from 'shared/util/constants';
import {COUNT, createOrderIOMap} from 'shared/util/pagination';
import {
	getMapResultToProps,
	mapPropsToOptions
} from 'contacts/hoc/mappers/interests-query';
import {graphql} from '@apollo/react-hoc';
import {Routes, toRoute} from 'shared/util/router';
import {useParams} from 'react-router-dom';
import {useQueryPagination} from 'shared/hooks';
import {withBaseResults} from 'shared/hoc';

const withData = () =>
	graphql(IndividualInterestsQuery, {
		options: mapPropsToOptions,
		props: getMapResultToProps(CompositionTypes.IndividualInterests)
	});

const TableWithData = withBaseResults(withData, {
	getColumns: ({channelId, groupId, maxCount, totalCount}) => [
		compositionListColumns.getName({
			label: Liferay.Language.get('topic'),
			maxWidth: 200,
			routeFn: ({data: {name}}) =>
				name &&
				toRoute(Routes.CONTACTS_INDIVIDUALS_INTEREST_DETAILS, {
					channelId,
					groupId,
					interestId: name
				}),
			sortable: true
		}),
		compositionListColumns.getRelativeMetricBar({
			label: Liferay.Language.get('total-individuals'),
			maxCount,
			sortable: true,
			totalCount
		}),
		compositionListColumns.getPercentOf({
			metricName: Liferay.Language.get('total-individuals'),
			totalCount
		})
	],
	rowIdentifier: 'name',
	showDropdownRangeKey: false
});

const Interests: React.FC<React.HTMLAttributes<HTMLElement>> = () => {
	const {channelId, groupId} = useParams();
	const {delta, orderIOMap, page, query} = useQueryPagination({
		initialOrderIOMap: createOrderIOMap(COUNT)
	});

	return (
		<Card pageDisplay>
			<Card.Header className='align-items-center d-flex justify-content-between'>
				<Card.Title>
					{Liferay.Language.get('interest-topics')}
				</Card.Title>
			</Card.Header>

			<TableWithData
				channelId={channelId}
				delta={delta}
				groupId={groupId}
				noResultsRenderer={
					<NoResultsDisplay
						description={
							<>
								{Liferay.Language.get(
									'check-back-later-to-verify-if-data-has-been-received-from-your-data-sources'
								)}

								<a
									className='d-block mb-3'
									href={
										URLConstants.IndividualsDashboardInterestsDocumentation
									}
									key='DOCUMENTATION'
									target='_blank'
								>
									{Liferay.Language.get(
										'learn-more-about-interests'
									)}
								</a>
							</>
						}
						icon={{
							border: false,
							size: Sizes.XXXLarge,
							symbol: 'ac-satellite'
						}}
						title={Liferay.Language.get(
							'there-are-no-interests-found'
						)}
					/>
				}
				orderIOMap={orderIOMap}
				page={page}
				query={query}
				rowBordered={false}
			/>
		</Card>
	);
};

export default Interests;
