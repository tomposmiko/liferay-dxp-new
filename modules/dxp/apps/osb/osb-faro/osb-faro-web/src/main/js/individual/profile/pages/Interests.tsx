import * as API from 'shared/api';
import Card from 'shared/components/Card';
import getCN from 'classnames';
import NoResultsDisplay from 'shared/components/NoResultsDisplay';
import React from 'react';
import SearchableEntityTable from 'shared/components/SearchableEntityTable';
import TextTruncate from 'shared/components/TextTruncate';
import URLConstants from 'shared/util/url-constants';
import {
	createOrderIOMap,
	getDefaultSortOrder,
	NAME
} from 'shared/util/pagination';
import {INDIVIDUALS} from 'shared/util/router';
import {interestListColumns} from 'shared/util/table-columns';
import {Routes, toRoute} from 'shared/util/router';
import {Sizes} from 'shared/util/constants';
import {sub} from 'shared/util/lang';
import {useQueryPagination} from 'shared/hooks';

export const TOTAL_DAYS = 90;

interface IContributionsCellProps {
	className?: string;
	data: {relatedPagesCount: number};
}

export const ContributionsCell: React.FC<IContributionsCellProps> = ({
	className,
	data: {relatedPagesCount}
}) => (
	<td className={getCN('table-cell-expand', className)}>
		<TextTruncate
			title={sub(Liferay.Language.get('x-contributing-pages'), [
				relatedPagesCount
			])}
		/>
	</td>
);

interface IInterestsProps {
	channelId: string;
	groupId: string;
	id: string;
}

const Interests: React.FC<IInterestsProps> = ({channelId, groupId, id}) => {
	const {delta, orderIOMap, page, query} = useQueryPagination({
		initialOrderIOMap: createOrderIOMap(NAME, getDefaultSortOrder(NAME))
	});

	return (
		<Card pageDisplay>
			<SearchableEntityTable
				className='interest-history-table'
				columns={[
					interestListColumns.getName({
						channelId,
						groupId,
						id,
						maxWidth: null,
						routeFn: ({data: {name}}) =>
							name &&
							toRoute(
								Routes.CONTACTS_INDIVIDUAL_INTEREST_DETAILS,
								{
									channelId,
									groupId,
									id,
									interestId: name
								}
							),
						type: INDIVIDUALS
					}),
					{
						accessor: 'relatedPagesCount',
						cellRenderer: ContributionsCell,
						label: Liferay.Language.get('contributing-pages'),
						sortable: false
					}
				]}
				dataSourceFn={API.interests.search}
				dataSourceParams={{
					contactsEntityId: id,
					groupId,
					interestMax: TOTAL_DAYS
				}}
				delta={delta}
				entityLabel={Liferay.Language.get('interests')}
				noResultsRenderer={() => (
					<NoResultsDisplay
						description={
							<>
								<span className='mr-1'>
									{Liferay.Language.get(
										'check-back-later-to-verify-if-data-has-been-received-from-your-data-sources'
									)}
								</span>

								<a
									href={
										URLConstants.IndividualsDashboardInterestsDocumentation
									}
									key='DOCUMENTATION'
									target='_blank'
								>
									{Liferay.Language.get(
										'learn-more-about-interests'
									)}
								</a>
							</>
						}
						icon={{
							border: false,
							size: Sizes.XXXLarge,
							symbol: 'ac-satellite'
						}}
						title={Liferay.Language.get(
							'there-are-no-interests-found'
						)}
					/>
				)}
				orderByOptions={[
					{
						label: Liferay.Language.get('interest'),
						value: NAME
					}
				]}
				orderIOMap={orderIOMap}
				page={page}
				query={query}
				rowIdentifier='name'
			/>
		</Card>
	);
};

export default Interests;
