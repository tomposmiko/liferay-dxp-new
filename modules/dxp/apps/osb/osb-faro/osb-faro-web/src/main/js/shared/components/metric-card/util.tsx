import MetricValue from './MetricValue';
import React, {Fragment} from 'react';
import Trend from 'shared/components/Trend';
import {CHART_COLOR_NAMES} from 'shared/components/Chart';
import {get, last} from 'lodash';
import {
	getAxisFormatter,
	getDataFormatter,
	getIntervals,
	getMetricFormatter
} from 'shared/util/charts';
import {getIcon, getStatsColor} from 'shared/util/metrics';
import {INTERVAL_KEY_MAP} from 'shared/util/time';
import {MetricType} from './metrics';
import {toRounded} from 'shared/util/numbers';
import {toUnix} from 'shared/util/date';

const CHART_DATA_ID_1 = 'data_1';
const CHART_DATA_ID_2 = 'data_2';
const CHART_DATA_PREVIOUS = 'data_previous';
const METRIC_TOOLTIP_LABEL_MAP = {
	bounceRateMetric: Liferay.Language.get('avg-bounce')
};

const {
	martell: CHART_GREEN,
	martellL2: CHART_GREEN_L2,
	mormont: CHART_ORANGE,
	stark: CHART_BLUE,
	starkL2: CHART_BLUE_L2
} = CHART_COLOR_NAMES;

export const Icons = {
	negative: 'caret-bottom-l',
	neutral: undefined,
	positive: 'caret-top-l'
};

const PREVIOUS_PERIOD_VISITORS_COLOR = '#393A4A';

type TBuildTabs = (props: {
	activeItemIndex: number;
	items: {
		content: {
			details: {
				color: string;
				icon: string;
				label: string;
			};
			title: string;
			type: MetricType;
			value: string;
		};
	}[];
	onActiveItemIndexChange: (index: number) => void;
}) => {
	onClick: () => void | ((index: number) => void);
	secondaryInfo: React.ReactElement;
}[];

export const buildTabs: TBuildTabs = ({
	activeItemIndex,
	items,
	onActiveItemIndexChange
}) =>
	items.map(({content}, index) => {
		const {details, title, type, value} = content;
		const {color, icon, label} = details;

		return {
			onClick: () => {
				if (activeItemIndex !== index && !onActiveItemIndexChange)
					return () => {};

				return onActiveItemIndexChange(index);
			},
			secondaryInfo: (
				<span>
					<span className='primary-content'>
						<MetricValue type={type} value={value} />
					</span>

					{label && <Trend color={color} icon={icon} label={label} />}
				</span>
			),
			tabId: index,
			title
		};
	});

export const getMetricName = (activeItemIndex, metrics) =>
	metrics.map(({name}) => name)[activeItemIndex];

export const getActiveItem = (retVal, compareToPrevious) => {
	if (!retVal) {
		return {
			chartData: [],
			intervals: [],
			timeline: []
		};
	}

	const chartData = retVal.data.slice(0, -1);
	const timeline = last(retVal.data);

	if (!compareToPrevious && retVal.asymmetricComparison) {
		retVal = {
			...retVal,
			chartData: chartData.map(dataSet => ({
				...dataSet,
				data: dataSet.data.slice(1)
			})),
			intervals: retVal.intervals.slice(1),

			// @ts-ignore

			timeline: {data: timeline.data.slice(1), id: timeline.id}
		};

		if (retVal.compositeData) {
			const compositeDataKeys = Object.keys(retVal.compositeData);

			retVal = {
				...retVal,
				compositeData: compositeDataKeys.reduce((acc, val) => {
					acc = {
						...acc,
						[val]: retVal.compositeData[val].slice(1)
					};

					return acc;
				}, {})
			};
		}
	} else if (compareToPrevious && retVal.asymmetricComparison) {
		retVal = {
			...retVal,
			chartData: chartData.map(dataSet => ({
				...dataSet,
				data:
					dataSet.id !== 'data_previous'
						? [null, ...dataSet.data.slice(1)]
						: dataSet.data
			})),
			timeline
		};
	} else {
		retVal = {
			...retVal,
			chartData,
			timeline
		};
	}

	return retVal;
};

export const getPreviousValueFromCompositeData = (
	compositeData,
	dataName,
	dateKey
) => {
	const data = get(compositeData, dataName);

	if (data) {
		return data.find(val => toUnix(val.key) === dateKey).previousValue;
	}
};

export const getRegexType = (type: MetricType): RegExp => {
	if (type === MetricType.Ratings) {
		return /([/][0-9]+)/g;
	} else {
		return /([a-zA-Z%])+/g;
	}
};

export const formatValue = (
	value: string,
	regex: RegExp
): React.ReactElement[] => {
	const items = value.split(' ');

	return items.map((item, i) => {
		const [value, unit] = item.split(regex);

		return (
			<Fragment key={i}>
				{value}

				<span className='metric-value-letter'>{unit}</span>
			</Fragment>
		);
	});
};

export const getMetricCardTabsData = (result, metrics) =>
	metrics.map(({name, title, type}) => {
		const metricFormatter = getMetricFormatter(type);

		return {
			content: {
				details: {
					color: getStatsColor(
						result[name].trend.trendClassification
					),
					icon: getIcon(result[name].trend.percentage),
					label: `${toRounded(
						Math.abs(result[name].trend.percentage)
					)}%`
				},
				name,
				title,
				type,
				value: metricFormatter(result[name].value)
			}
		};
	});

export const getMetricsData = (
	result,
	metrics,
	rangeSelectors = {},
	chartDataMapFn = getMetricsChartData,
	interval = INTERVAL_KEY_MAP.day
) =>
	metrics.map(({compositeMetrics, name, title, tooltipTitle, type}) =>
		getMetricData({
			chartDataMapFn,
			compositeMetrics,
			interval,
			name,
			rangeSelectors,
			result,
			title,
			tooltipTitle,
			type
		})
	);

export const getMetricsChartData = ({
	histogram,
	name,
	title,
	tooltipTitle,
	type
}) => [
	{
		color: CHART_BLUE,
		data: getDataFormatter(type)(histogram.map(({value}) => value)),
		id: CHART_DATA_ID_1,
		name: tooltipTitle || METRIC_TOOLTIP_LABEL_MAP[name] || title,
		tooltipTitle
	},
	{
		color: CHART_BLUE_L2,
		data: getDataFormatter(type)(
			histogram.map(({previousValue}) => previousValue)
		),
		id: CHART_DATA_PREVIOUS,
		name: Liferay.Language.get('previous-period')
	},
	{
		data: histogram.map(({key}) => key),
		id: 'x'
	}
];

export const convertHistogramKeysToDate = ({
	key,
	previousValueKey,
	valueKey,
	...otherParams
}) => ({
	key: toUnix(key),
	previousValueKey: previousValueKey.split('/').map(toUnix),
	valueKey: valueKey.split('/').map(toUnix),
	...otherParams
});

export const getMetricData = ({
	chartDataMapFn = getMetricsChartData,
	compositeMetrics,
	interval = INTERVAL_KEY_MAP.day,
	name,
	rangeSelectors,
	result,
	title,
	tooltipTitle,
	type
}) => {
	const metricFormatter = getMetricFormatter(type);

	const histogram = result[name].histogram.metrics.map(
		convertHistogramKeysToDate
	);

	const compositeData = compositeMetrics
		? {
				compositeContent: compositeMetrics.reduce(
					(
						acc,
						{
							name: compositeMetricName,
							title: compositeMetricTitle,
							type: compositeMetricType
						}
					) => {
						acc[compositeMetricName] = {
							details: {
								color: getStatsColor(
									result[compositeMetricName].trend
										.trendClassification
								),
								icon: getIcon(
									result[compositeMetricName].trend.percentage
								),
								label: `${toRounded(
									Math.abs(
										result[compositeMetricName].trend
											.percentage
									)
								)}%`
							},
							name: compositeMetricName,
							title: compositeMetricTitle,
							type: compositeMetricType,
							value: metricFormatter(
								result[compositeMetricName].value
							)
						};

						return acc;
					},
					{}
				),
				compositeData: compositeMetrics.reduce(
					(acc, {name: compositeMetricName}) => {
						acc[compositeMetricName] =
							result[compositeMetricName].histogram.metrics;

						return acc;
					},
					{}
				)
		  }
		: {};

	const dateKeysIMap = new Map(
		histogram.map(({key, valueKey}) => [key, valueKey])
	);

	return {
		...compositeData,
		asymmetricComparison: result[name].histogram.asymmetricComparison,
		content: {
			details: {
				color: getStatsColor(result[name].trend.trendClassification),
				icon: getIcon(result[name].trend.percentage),
				label: `${toRounded(Math.abs(result[name].trend.percentage))}%`
			},
			name,
			title,
			type,
			value: metricFormatter(result[name].value)
		},
		data: chartDataMapFn({
			...compositeData,
			histogram,
			name,
			title,
			tooltipTitle,
			type
		}),
		dateKeysIMap,
		format: getAxisFormatter(type),
		intervals: getIntervals(
			rangeSelectors.rangeKey,
			histogram.map(({key}) => key),
			interval,
			dateKeysIMap
		),
		prevDateKeysIMap: new Map(
			histogram.map(({key, previousValueKey}) => [key, previousValueKey])
		)
	};
};

export const getSiteMetricsChartData = ({
	compositeData,
	histogram,
	name,
	title,
	tooltipTitle,
	type
}) =>
	name === 'visitorsMetric'
		? [
				{
					color: CHART_BLUE,
					data: getDataFormatter(type)(
						compositeData.knownVisitorsMetric.map(
							({value}) => value
						)
					),
					dataName: 'knownVisitorsMetric',
					id: CHART_DATA_ID_1,
					name: Liferay.Language.get('known-visitors'),
					tooltipTitle: Liferay.Language.get('known'),
					type: 'bar'
				},
				{
					color: CHART_ORANGE,
					data: getDataFormatter(type)(
						compositeData.anonymousVisitorsMetric.map(
							({value}) => value
						)
					),
					dataName: 'anonymousVisitorsMetric',
					id: CHART_DATA_ID_2,
					name: Liferay.Language.get('anonymous-visitors'),
					tooltipTitle: Liferay.Language.get('anonymous'),
					type: 'bar'
				},
				{
					color: PREVIOUS_PERIOD_VISITORS_COLOR,
					data: getDataFormatter(type)(
						histogram.map(({previousValue}) => previousValue)
					),
					id: CHART_DATA_PREVIOUS,
					name: Liferay.Language.get('previous-period'),
					type: 'line'
				},
				{
					data: histogram.map(({key}) => key),
					id: 'x'
				}
		  ]
		: getMetricsChartData({histogram, name, title, tooltipTitle, type}).map(
				data =>
					[CHART_DATA_ID_1, CHART_DATA_PREVIOUS].includes(data.id)
						? {
								...data,
								color:
									data.id === CHART_DATA_PREVIOUS
										? CHART_GREEN_L2
										: CHART_GREEN
						  }
						: data
		  );
