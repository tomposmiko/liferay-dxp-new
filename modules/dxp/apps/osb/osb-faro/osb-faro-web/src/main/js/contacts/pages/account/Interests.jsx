import Interests from 'contacts/hoc/account/Interests';
import React from 'react';
import {PropTypes} from 'prop-types';

export default class InterestsPage extends React.Component {
	static propTypes = {
		router: PropTypes.object
	};

	render() {
		const {router} = this.props;

		return (
			<div className='account-interests-root'>
				<div className='row'>
					<div className='col-xl-12'>
						<Interests router={router} />
					</div>
				</div>
			</div>
		);
	}
}
