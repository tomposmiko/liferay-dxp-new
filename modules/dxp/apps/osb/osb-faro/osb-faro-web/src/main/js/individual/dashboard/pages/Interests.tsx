import BasePage from 'shared/components/base-page';
import ClayLink from '@clayui/link';
import Interests from '../hocs/Interests';
import React from 'react';
import StatesRenderer from 'shared/components/states-renderer/StatesRenderer';
import URLConstants from 'shared/util/url-constants';
import {Routes, toRoute} from 'shared/util/router';
import {useDataSource} from 'shared/hooks/useDataSource';
import {useParams} from 'react-router-dom';
import {User} from 'shared/util/records';
import {withCurrentUser} from 'shared/hoc';

interface IInterestsPageProps extends React.HTMLAttributes<HTMLElement> {
	currentUser: User;
}

const InterestsPage: React.FC<IInterestsPageProps> = ({currentUser}) => {
	const {groupId} = useParams();
	const authorized = currentUser.isAdmin();

	const dataSourceStates = useDataSource();

	return (
		<BasePage.Body pageContainer>
			<StatesRenderer {...dataSourceStates}>
				<StatesRenderer.Empty
					description={
						<>
							{Liferay.Language.get(
								'connect-a-data-source-with-sites-data'
							)}

							<a
								className='d-block mb-3'
								href={URLConstants.DataSourceConnection}
								key='DOCUMENTATION'
								target='_blank'
							>
								{Liferay.Language.get(
									'access-our-documentation-to-learn-more'
								)}
							</a>

							{authorized && (
								<ClayLink
									button
									className='button-root'
									displayType='primary'
									href={toRoute(
										Routes.SETTINGS_ADD_DATA_SOURCE,
										{
											groupId
										}
									)}
								>
									{Liferay.Language.get(
										'connect-data-source'
									)}
								</ClayLink>
							)}
						</>
					}
					displayCard
					title={Liferay.Language.get(
						'no-sites-synced-from-data-sources'
					)}
				/>

				<StatesRenderer.Success>
					<div className='individuals-dashboard-interests-root'>
						<Interests />
					</div>
				</StatesRenderer.Success>
			</StatesRenderer>
		</BasePage.Body>
	);
};

export default withCurrentUser(InterestsPage);
