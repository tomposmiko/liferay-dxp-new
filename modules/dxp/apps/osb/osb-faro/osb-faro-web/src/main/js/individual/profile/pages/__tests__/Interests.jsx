import Interests, {ContributionsCell} from '../Interests';
import React from 'react';
import {MemoryRouter, Route} from 'react-router-dom';
import {render, waitForElementToBeRemoved} from '@testing-library/react';
import {Routes} from 'shared/util/router';

jest.unmock('react-dom');

describe('Interests', () => {
	it('should render', async () => {
		const {container} = render(
			<MemoryRouter
				initialEntries={[
					'/workspace/23/123/contacts/individuals/known-individuals/321321/interests?delta=2&page=1&field=count&sortOrder=DESC'
				]}
			>
				<Route path={Routes.CONTACTS_INDIVIDUAL_INTERESTS}>
					<Interests channelId='123' groupId='23' id='test' />
				</Route>
			</MemoryRouter>
		);

		await waitForElementToBeRemoved(
			container.querySelector('.spinner-root')
		);

		jest.runAllTimers();

		expect(container).toMatchSnapshot();
	});
});

describe('ContributionsCell', () => {
	it('should render', () => {
		const {container} = render(
			<ContributionsCell data={{relatedPagesCount: 8}} />
		);
		expect(container).toMatchSnapshot();
	});
});
