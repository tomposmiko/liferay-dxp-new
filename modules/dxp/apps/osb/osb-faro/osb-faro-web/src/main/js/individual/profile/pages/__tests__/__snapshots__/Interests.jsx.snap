// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributionsCell should render 1`] = `
<div>
  <td
    class="table-cell-expand"
  >
    <span
      class="text-truncate"
      data-tooltip="false"
      title=""
    >
      8 Contributing Pages
    </span>
  </td>
</div>
`;

exports[`Interests should render 1`] = `
<div>
  <div
    class="card card-root page-display"
  >
    <div
      class="base-results-root d-flex flex-column flex-grow-1 searchable-table-root interest-history-table searchable-table-borderless"
    >
      <div
        class="toolbar-root"
      >
        <nav
          class="navbar navbar-root management-bar management-bar-light navbar-expand"
        >
          <div
            class="container-fluid"
          >
            <ul
              class="nav-root front-nav navbar-nav"
              role="tablist"
            >
              <li
                aria-haspopup="true"
                class="dropdown dropdown-root filter-and-order-root nav-item"
              >
                <button
                  class="button-root btn btn-unstyled dropdown-toggle nav-btn show-caret"
                  data-testid="filter-and-order-button"
                  type="button"
                >
                  <span
                    class="text-truncate"
                  >
                    Order
                  </span>
                  
                  <span
                    class="caret-root"
                  >
                    <svg
                      class="icon-root lexicon-icon lexicon-icon-caret-bottom"
                      viewBox=""
                    >
                      <use
                        xlink:href="/o/osb-faro-web/dist/sprite.svg#caret-bottom"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="nav-item"
                role="presentation"
              >
                <a
                  class="button-root btn btn-unstyled nav-link nav-link-monospaced btn-outline-borderless"
                  href="/?field=count&page=1&sortOrder=ASC"
                >
                  <svg
                    class="icon-root lexicon-icon lexicon-icon-order-arrow-descending"
                    viewBox=""
                  >
                    <use
                      xlink:href="/o/osb-faro-web/dist/sprite.svg#order-arrow-descending"
                    />
                  </svg>
                </a>
              </li>
            </ul>
            <div
              class="navbar-form navbar-form-autofit"
            >
              <div
                class="input-group search"
              >
                <div
                  class="input-group-item"
                >
                  <input
                    class="input-root form-control input-group-inset input-group-inset-after"
                    placeholder="Search"
                    type="text"
                    value=""
                  />
                  <div
                    class="input-group-inset-item input-group-inset-item-after"
                  >
                    <button
                      class="button-root btn btn-unstyled"
                      type="button"
                    >
                      <svg
                        class="icon-root lexicon-icon lexicon-icon-search"
                        viewBox=""
                      >
                        <use
                          xlink:href="/o/osb-faro-web/dist/sprite.svg#search"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </div>
      <div
        class="table-responsive table-root flex-grow-1"
      >
        <table
          class="table table-autofit table-hover table-heading-nowrap table-nowrap"
        >
          <thead>
            <tr>
              <th
                class="table-head-title"
              >
                <a
                  class="button-root btn btn-unstyled inline-item text-truncate-inline"
                  href="/?field=name&page=1&sortOrder=ASC"
                >
                  <span
                    class="text-truncate"
                  >
                    Interest
                  </span>
                </a>
              </th>
              <th
                class="table-head-title"
              >
                Contributing Pages
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class=""
            >
              <td
                class="name-cell-root"
              >
                <div
                  class="content-container"
                >
                  <div
                    class="text-truncate"
                  >
                    <div
                      class="table-title text-truncate"
                    >
                      <a
                        class="text-truncate"
                        href="/workspace/23/123/contacts/individuals/known-individuals/test/interests/Bar0"
                      >
                        Bar0
                      </a>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="table-cell-expand"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  8 Contributing Pages
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <nav
        class="navbar navbar-root pagination-bar-root"
      >
        <div
          class="container-fluid"
        >
          <div
            aria-haspopup="true"
            class="dropdown dropdown-root pagination-items-per-page"
          >
            <button
              class="button-root btn btn-secondary dropdown-toggle show-caret btn-sm"
              type="button"
            >
              <span
                class="text-truncate"
              >
                2 Items
              </span>
              
              <span
                class="caret-root"
              >
                <svg
                  class="icon-root lexicon-icon lexicon-icon-caret-bottom"
                  viewBox=""
                >
                  <use
                    xlink:href="/o/osb-faro-web/dist/sprite.svg#caret-bottom"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="pagination-results"
          >
            Showing 1 to 1 of 1 entry.
          </div>
          <ul
            class="pagination pagination-root"
          >
            <li
              class="page-item disabled"
            >
              <a
                class="button-root btn btn-unstyled page-link link-disabled"
              >
                <svg
                  class="icon-root lexicon-icon lexicon-icon-angle-left"
                  viewBox=""
                >
                  <use
                    xlink:href="/o/osb-faro-web/dist/sprite.svg#angle-left"
                  />
                </svg>
              </a>
            </li>
            <li
              class="page-item active"
            >
              <a
                class="button-root btn btn-unstyled page-link"
                href="/?page=1"
              >
                1
              </a>
            </li>
            <li
              class="page-item disabled"
            >
              <a
                class="button-root btn btn-unstyled page-link link-disabled"
              >
                <svg
                  class="icon-root lexicon-icon lexicon-icon-angle-right"
                  viewBox=""
                >
                  <use
                    xlink:href="/o/osb-faro-web/dist/sprite.svg#angle-right"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
